java/code/src/com/redhat/rhn/manager/formula/FormulaManager.java

```diff
@@ -342,4 +342,48 @@ public void enableFormula(String minionId, String formulaName) throws IOExceptio
             FormulaFactory.saveServerFormulas(minionId, enabledFormulas);
         }
     }
+
+    /**
+     * Configure system for monitoring when the entitlement is added.
+     * @param minion the MinionServer to be configured for monitoring
+     */
+    public void enableMonitoringOnEntitlementAdd(MinionServer minion) {
+        try {
+            // Assign the monitoring formula to the system unless it belongs to a group having monitoring enabled
+            if (!FormulaFactory.isMemberOfGroupHavingMonitoring(minion)) {
+                List<String> formulas = FormulaFactory.getFormulasByMinionId(minion.getMinionId());
+                if (!formulas.contains(FormulaFactory.PROMETHEUS_EXPORTERS)) {
+                    formulas.add(FormulaFactory.PROMETHEUS_EXPORTERS);
+                    FormulaFactory.saveServerFormulas(minion.getMinionId(), formulas);
+                }
+            }
+        }
+        catch (UnsupportedOperationException | IOException e) {
+            // FIXME: error handling
+            // LOG.error("Error assigning formula: " + e.getMessage(), e);
+            // result.addError(new ValidatorError("system.entitle.formula_error"));
```

Comment:
i think to be consistent with the previous code this would need to return the `ValidatorError` and the places using it would need to merge the this `ValidatorError` with the one returned by `addEntitlementToServer`.

Reply:
Right, thanks for the input, I will work on it.

Reply:
@lucidd eventually I ended up doing some more consistent error handling: while the `UnsupportedOperationException` seems to be obsolete here meanwhile I am now throwing the `IOException` so it can be handled in the calling code appropriately. When the call comes from the UI there will be a message displayed, otherwise the error will be written to the logs.